crates/sdk/src/build.rs

```diff
@@ -697,6 +723,176 @@ impl AxiomSdk {
             ))
         }
     }
+
+    pub fn upload_elf_base(
+        &self,
+        program_dir: impl AsRef<Path>,
+        args: UploadElfArgs,
+        callback: &dyn ProgressCallback,
+    ) -> Result<String> {
+        // Check if we're in a Rust project
+        if !is_rust_project(program_dir.as_ref()) {
+            eyre::bail!("Not in a Rust project. Make sure Cargo.toml exists.");
+        }
+
+        // Look for the openvm build output directory
+        let target_dir = program_dir.as_ref().join("target").join("openvm");
+        if !target_dir.exists() {
+            eyre::bail!(
+                "OpenVM build output not found. Please run 'cargo openvm build' first.\nExpected directory: {}",
+                target_dir.display()
+            );
+        }
+
+        // Find the release directory
+        let release_dir = target_dir.join("release");
+        if !release_dir.exists() {
+            eyre::bail!(
+                "OpenVM release build not found. Please run 'cargo openvm build' first.\nExpected directory: {}",
+                release_dir.display()
+            );
+        }
+
+        // Find ELF and VMEXE files
+        let elf_files: Vec<_> = std::fs::read_dir(&release_dir)?
+            .filter_map(Result::ok)
+            .filter(|entry| entry.path().extension().and_then(|s| s.to_str()) == Some("elf"))
+            .collect();
+
+        if elf_files.is_empty() {
+            eyre::bail!(
+                "No ELF files found in {}. Please run 'cargo openvm build' first.",
+                release_dir.display()
+            );
+        }
+
+        if elf_files.len() > 1 {
```

Comment:
it's pretty common for a project to have multiple `bin/` in which case there will be multiple ELF binaries. shouldn't you filter on binary name?